File: TUnit.TestProject.FSharp/NullableByteArgumentTests.fs
namespace TUnit.TestProject.FSharp

open TUnit.Core

// F# equivalent of NullableByteArgumentTests.cs

type NullableByteArgumentTests() =
    [<Test>]
    [<Arguments(1uy)>]
    [<Arguments(null)>]
    member _.Test(someByte: byte option) =
        ()

    [<Test>]
    [<Arguments(1uy, 1uy)>]
    [<Arguments(1uy, null)>]
    member _.Test2(byte1: byte, byte2: byte option) =
        ()

File: TUnit.TestProject/NullableByteArgumentTests.cs
using TUnit.TestProject.Attributes;

namespace TUnit.TestProject;

[EngineTest(ExpectedResult.Pass)]
public class NullableByteArgumentTests
{
    [Test]
    [Arguments((byte)1)]
    [Arguments(null)]
    public void Test(byte? someByte)
    {
    }

    [Test]
    [Arguments((byte)1, (byte)1)]
    [Arguments((byte)1, null)]
    public void Test2(byte byte1, byte? byte2)
    {
    }
}